dali/kernels/signal/resampling.h

```diff
@@ -132,11 +181,27 @@ struct Resampler {
         std::tie(i0, i1) = window.input_range(in_pos);
         if (i0 + in_block_i < 0)
           i0 = -in_block_i;
-        if (i1 + in_block_i >= n_in)
-          i1 = n_in - 1 - in_block_i;
+        if (i1 + in_block_i > n_in)
+          i1 = n_in - in_block_i;
         float f = 0;
         int i = i0;
 
+#ifdef __ARM_NEON
+        float32x4_t f4 = vdupq_n_f32(0);
+
+        float32x4_t x4 = vaddq_f32(vdupq_n_f32(i - in_pos), _0123);
+
+        for (; i + 3 <= i1; i += 4) {
+            float32x4_t w4 = window(x4);
+            f4 = vfmaq_f32(f4, vld1q_f32(in_block_ptr + i), w4);
+            x4 = vaddq_f32(x4, vdupq_n_f32(4));
+        }
+        // Reduce elements in f4
```

Comment:
You can add the same comment in L214 - it was unclear for me why is happening there.